webapp/src/modules/item/reducer.spec.ts

```diff
@@ -207,19 +230,42 @@ describe.each([
 
 describe('when reducing the successful action of fetching an item', () => {
   const requestAction = fetchItemRequest(item.contractAddress, item.itemId)
-  const successAction = fetchItemSuccess(item)
+  let successAction = fetchItemSuccess(item)
 
-  const initialState = {
+  let initialState = {
     ...INITIAL_STATE,
     data: { anotherId: anotherItem },
     loading: loadingReducer([], requestAction)
   }
 
-  it('should return a state with the the loaded items with the fetched item and the loading state cleared', () => {
-    expect(itemReducer(initialState, successAction)).toEqual({
-      ...INITIAL_STATE,
-      loading: [],
-      data: { ...initialState.data, [item.id]: item }
+  describe('and the fetched item is not in the state', () => {
+    it('should return a state with the the loaded items with the fetched item and the loading state cleared', () => {
+      expect(itemReducer(initialState, successAction)).toEqual({
+        ...INITIAL_STATE,
+        loading: [],
+        data: { ...initialState.data, [item.id]: item }
+      })
+    })
+  })
+
+  describe('and the item is already in the state', () => {
+    const newItemData = {
+      minPrice: '1234'
+    }
+    beforeEach(() => {
+      initialState = {
+        ...INITIAL_STATE,
+        data: { anotherId: anotherItem, [item.id]: item },
+        loading: loadingReducer([], requestAction)
+      }
+      successAction = fetchItemSuccess({ ...item, ...newItemData })
+    })
```

Comment:
Would you mind defining the newItemData inside of the before each?
```suggestion
    let newItemData: Item
    beforeEach(() => {
      newItemData = {
        minPrice: '1234'
      } as Item
      initialState = {
        ...INITIAL_STATE,
        data: { anotherId: anotherItem, [item.id]: item },
        loading: loadingReducer([], requestAction)
      }
      successAction = fetchItemSuccess({ ...item, ...newItemData })
    })
```

Reply:
sure, done!